### ISSUE TITLE
ws: Add `autoReconnect` option?

### ISSUE THREAD

**Issue by crishoj** [elysiajs/eden#107]
After migrating from VueUse `useWebSocket`, I realized that EdenWS doesn't have an auto-reconnect option.

Would you welcome a PR adding an `autoReconnect` option?

**Comment by dandeduck** [comment]
Makes sense, if you ask me, ws should accept the same options as the underlying ws implementation.

**Comment by crishoj** [comment]
The auto-reconnect is not actually present in the underlying (bare) [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket), but since handling disconnects (by reconnecting) is a fairly common need, I think it could be warranted to add in Eden.

**Comment by dandeduck** [comment]
ahh right, I think it would be welcome if you created a PR with it (But I'm not the maintainer or anything, I'll be happy to give you a review, but I cannot merge it of course)
